Application/EdFi.Ods.AdminApi/Infrastructure/Database/Commands/AddTenantCommand.cs

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using EdFi.Ods.AdminApi.Infrastructure.Helpers;
+
+namespace EdFi.Ods.AdminApi.Infrastructure.Database.Commands;
+
+public class AddTenantCommand(IAppSettingsFileProvider fileProvider)
+{
+    private static readonly object _fileLock = new();
+    private readonly IAppSettingsFileProvider _fileProvider = fileProvider;
+
+    public virtual void Execute(IAddTenantModel model)
+    {
+        lock (_fileLock)
+        {
+            var json = _fileProvider.ReadAllText();
+            try
+            {
+                var root = JsonNode.Parse(json) ?? throw new InvalidOperationException("appsettings.json is empty or invalid.");
```

Comment:
@DavidJGapCR please revert this change. We cannot let the application rewrite the settings file.